invenio_app_ils/ui/src/invenio_app_ils/pages/frontsite/BookDetails/components/BookMetadata/components/RequestNewLoanForm/RequestNewLoanForm.js

```diff
@@ -0,0 +1,46 @@
+import React, { Component } from 'react';
+import { Form, Transition } from 'semantic-ui-react';
+
+export default class RequestNewLoanForm extends Component {
+  constructor(props) {
+    super(props);
+    this.requestNewLoanForBook = this.props.requestNewLoanForBook.bind(this);
+  }
+
+  requestNewLoan = (itemPid, loan) => {
+    this.requestNewLoanForBook(itemPid, loan);
+  };
+
+  render() {
+    const { docPid, visible } = this.props;
+
+    const loanData = {
+      document_pid: docPid,
+      patron_pid: '111', // FIXME: change this to be able to search for the user
+    };
+    const loanPeriodOptions = [
+      { key: '1', value: '1', text: '2 Weeks' },
+      { key: '2', value: '2', text: '3 Weeks' },
+    ];
+
+    return (
+      <Transition visible={visible} animation="scale" duration={500}>
+        <Form>
```

Comment:
@ntarocco do you think it is worth it to use https://www.npmjs.com/package/react-jsonschema-form here?

Reply:
shall we chat about that? :) :) :)